### PR TITLE
Calldwell and integration tests cleanup

### PR DESCRIPTION
**Pull request by SteelPh0enix**
- Moved common Calldwell code to proper helper modules and `calldwell-rs`
- Added script for building HAL integration tests
- Added CI step for building HAL integration tests
- General cleanup

### CODE REVIEW

File: calldwell/py.typed
Comment (Glamhoth) [inline calldwell/py.typed:1]: Empty file

File: calldwell/rust_helpers.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
+"""Module containing some boilerplate, common to all tests that use `calldwell-rs` library."""
+import logging
+from typing import Any, Callable, Optional, Tuple
+
+from .gdb_client import GDBClient
+from .rtt_client import RTTClient
+
+RTT_SECTION_SYMBOL_NAME = "_SEGGER_RTT"
+"""Section name of RTT symbol. Hard-coded in `rtt_target` library."""
```
Comment (Glamhoth) [inline calldwell/rust_helpers.py:9]: Shouldn't the docs be above the line they are documenting?

**Review by Glamhoth — CHANGES_REQUESTED** [review]

File: calldwell/__init__.py
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply calldwell/__init__.py:1]: It should be here, as it's a special Python file marking this directory as module.

File: calldwell/py.typed
Comment (SteelPh0enix) [reply calldwell/py.typed:1]: It should be here, to mark this directory as one containing files with type hints.
See https://peps.python.org/pep-0561/

File: calldwell/__init__.py
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply calldwell/__init__.py:1]: See: https://docs.python.org/3/tutorial/modules.html

File: calldwell/rust_helpers.py (hunk 1 above)
Comment (SteelPh0enix) [reply calldwell/rust_helpers.py:9]: Not in Python

**Review by Glamhoth — APPROVED** [review]

**Pull request merged**